apps/web/pages/api/book/event.ts

```diff
@@ -508,7 +508,27 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       }
     }
 
-    return prisma.booking.create(createBookingObj);
+    /*Validate if there is any stripe_payment credential for this user*/
+    const eventTypePayment = await prisma.credential.findFirst({
+      where: {
+        type: "stripe_payment",
+        userId: users[0].id,
+      },
+      select: {
+        id: true,
+      },
+    });
+    /** If the stripe_payment credential exists then it creates the booking OR*/
+    /** If the stripe_payment credential does not exist but the eventTypes is
+     *   without payment then create the booking
+     */
+    if ((eventTypePayment && eventType.price > 0) || !eventType.price) {
+      return prisma.booking.create(createBookingObj);
+    }
+    /** If the stripe_payment credential does not exist and the eventTypes is
```

Comment:
single line comment its enough here
```suggestion
// stripePaymentCredential not found and eventType requires payment we return null
```